### PR TITLE
Add workaround for Kafka Confluent Schema Registry error on startup

### PR DESCRIPTION
**Pull request by tlblessing**
## Description

Document a manual workaround per @hashhar for Kafka Confluent Schema Registry error on startup.

## Additional context and related issues

https://github.com/trinodb/trino/issues/17299

### CODE REVIEW

File: docs/src/main/sphinx/connector/kafka.rst [hunk 1]
```diff
@@ -420,6 +422,18 @@ table description supplier are:
 * There is no need to define tables manually.
 * Some Protobuf specific types like ``oneof`` are supported and mapped to JSON.
 
+When using Protobuf decoder with the Confluent table description supplier, the
+following additional steps must be taken:
+
+- Copy the JARs ``kafka-protobuf-provider`` and ``kafka-protobuf-types`` for
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/kafka.rst:428]: Where do you get those from?

File: docs/src/main/sphinx/connector/kafka.rst [hunk 2]
```diff
@@ -420,6 +422,18 @@ table description supplier are:
...
+  Confluent version 7.3.1 to the Kafka connector plugin directory on all nodes in
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/kafka.rst:429]: what directory is this? /plugin/kafka

File: docs/src/main/sphinx/connector/kafka.rst [hunk 3]
```diff
@@ -420,6 +422,18 @@ table description supplier are:
...
+  the cluster.
+- The JARs are provided under the Confluent license by Confluent. Accept those
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/kafka.rst:431]: This needs to be first and have details on how to get them from Confluent .. or link to their site or whatever

**Review by mosabua — CHANGES_REQUESTED** [review]

File: docs/src/main/sphinx/connector/kafka.rst (hunk 2 above)
Comment (mosabua) [reply docs/src/main/sphinx/connector/kafka.rst:429]: And maybe we need a section in the kubnertes page on how to add that file with helm (if that is even possible at the moment with the trino chart)

File: docs/src/main/sphinx/connector/kafka.rst (hunk 1 above)
Comment (hashhar) [reply docs/src/main/sphinx/connector/kafka.rst:428]: https://packages.confluent.io/maven/io/confluent/

File: docs/src/main/sphinx/connector/kafka.rst (hunk 3 above)
Comment (tlblessing) [reply docs/src/main/sphinx/connector/kafka.rst:432]: Added link to first bullet point. https://github.com/trinodb/trino/pull/17924#discussion_r1232059940

The order is as recommended by Ashhar and makes sense to me.

Comment (mosabua) [reply docs/src/main/sphinx/connector/kafka.rst:432]: How do you accept the license? The download is wide open so there is not license to accept.

**Comment by mosabua** [comment]
Please squash commits and address feedback.

File: docs/src/main/sphinx/connector/kafka.rst (hunk 3 above)
Comment (hashhar) [reply docs/src/main/sphinx/connector/kafka.rst:432]: You accept license by agreeing to use those things. It's a bit hostile but that's how it is.
The pom for each of them includes the license under which they are released (Confluent Community License).

Comment (hashhar) [reply docs/src/main/sphinx/connector/kafka.rst:432]: Can probably be reworded as "By copying those JARs and using them you agree to the terms of the license under which Confluent makes them available"

File: docs/src/main/sphinx/connector/kafka.rst (hunk 2 above)
Comment (tlblessing) [reply docs/src/main/sphinx/connector/kafka.rst]: @hashhar is it possible to add the jar files with Helm? Per Manfred's comment above.

**Comment by tlblessing** [comment]
> Please squash commits and address feedback.

I do not have all of the answers to your questions/suggestions yet.

File: docs/src/main/sphinx/connector/kafka.rst (hunk 2 above)
Comment (mosabua) [reply docs/src/main/sphinx/connector/kafka.rst]: Maybe just keep the mention generic that it needs to go into the plugin/kafka folder of the installation directory of Trino. And that folder varies between tarball, rpm, and container.

Comment (hashhar) [reply docs/src/main/sphinx/connector/kafka.rst]: Helm is not unique. You add JARs like how you add manually. You place them in a directory and mount that to the container and copy those. However the recommended practice with Docker is to create your own containers and morever this page is not a deployment specific page. We don't want to have instructions for Helm/k8s, OpenShift, Rancher and so on - it's up to whoever is deploying to know how to "Copy a file".

Comment (hashhar) [reply docs/src/main/sphinx/connector/kafka.rst]: > And that folder varies between tarball, rpm, and container.

If you say `plugin/kafka` then that folder doesn't change ever - only it's parent location might change - and that's why we don't use absolute paths in docs.

File: docs/src/main/sphinx/connector/kafka.rst (hunk 3 above)
Comment (hashhar) [reply docs/src/main/sphinx/connector/kafka.rst:432]: maybe add a link to https://github.com/confluentinc/schema-registry/blob/master/LICENSE-ConfluentCommunity - which is where the JARs come from and that's their license.

File: docs/src/main/sphinx/connector/kafka.rst [hunk 4]
```diff
@@ -420,6 +422,19 @@ table description supplier are:
 * There is no need to define tables manually.
 * Some Protobuf specific types like ``oneof`` are supported and mapped to JSON.
 
+When using Protobuf decoder with the Confluent table description supplier, the
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/kafka.rst:425]: I thought this over some more and checked with other docs .. we should move all this up to the requirements section and then link to it from here

Comment (tlblessing) [reply docs/src/main/sphinx/connector/kafka.rst]: Moved new material to requirements section and linked.

**Comment by tlblessing** [comment]
@mosabua and @hashhar the PR is ready for another review. Thank you for your guidance.

**Review by mosabua — APPROVED** [review]
Looks good now.

**Pull request merged**